lib/bugzilla_service.rb

```diff
@@ -13,26 +13,126 @@ class BugzillaService
     resolved
   )
 
+  # Helper class for specific bugs
+  class Bug
+    attr_reader   :service
+    attr_accessor :id, :status
+
+    def initialize(service, id, status)
+      @service, @id, @status = service, id, status
+    end
+
+    def comments
+      response = service.get("/rest/bug/#{id}/comment") do |req|
+        req.params["include_fields"] = "text"
+      end
+
+      if response.status == 200
+        JSON.parse(response.body)["bugs"][id.to_s]["comments"]
+            .map { |comment| comment["text"] }
+      else
+        []
+      end
+    end
+
+    # Adds a comment.  Returns true if status == 200
+    def add_comment(text)
+      payload  = {"comment" => text}.to_json
+      response = service.post("/rest/bug/#{id}/comment", payload)
+
+      response.status == 200
+    end
+
+    # Basicially API compatible with `active_bugzilla` way of updating the bug
+    # status.
+    def save
+      payload  = { "ids" => [id], "status" => status }.to_json
+      response = service.put("/rest/bug/#{id}", payload)
+
+      response.status == 200
+    end
+  end
+
   def initialize
-    service # initialize the service
+    connection # initialize the connection
   end
 
-  def service
-    @service ||= begin
-      require 'active_bugzilla'
-      bz = ActiveBugzilla::Service.new(*credentials.to_h.values_at(:url, :username, :password))
-      ActiveBugzilla::Base.service = bz
+  def connection
+    @connection ||= begin
+      require 'faraday'
+
+      Faraday.new(:url => credentials.to_h[:url]) do |faraday|
+        faraday.request(:url_encoded)
+        faraday.adapter(Faraday.default_adapter)
+      end
     end
   end
 
   def find_bug(id)
-    ActiveBugzilla::Bug.find(:product => Settings.bugzilla.product, :id => id).first
+    response = get("/rest/bug") do |req|
+      product = Settings.bugzilla.product
+      req.params["id"]             = id
+      req.params["product"]        = product if product
+      req.params["include_fields"] = "id,status"
+    end
+
+    if response.status == 200
+      attributes = JSON.parse(response.body)["bugs"].first
+      Bug.new(self, attributes['id'], attributes['status'])
+    end
   end
 
   def with_bug(id)
     yield find_bug(id)
   end
 
+  # Shared request helpers for adding BZ auth to a request
+  def get(path, &block)
+    connection.get(&faraday_request_block(path, &block))
+  end
+
+  def post(path, payload, &block)
+    connection.post(&put_post_block(path, payload, &block))
+  end
+
+  def put(path, payload, &block)
+    connection.put(&put_post_block(path, payload, &block))
+  end
+
+  private
+
+  # Shared request block for adding the URL and block (required for both GET
+  # and POST requests)
+  def faraday_request_block(path)
+    credentials = [
+      Settings.bugzilla_credentials.username,
+      Settings.bugzilla_credentials.password
+    ].compact
+
+    lambda do |req|
+      req.url(path)
+      unless credentials.empty?
```

Comment:
We probably should check that both username and password are set, else blow up, a la https://github.com/ManageIQ/active_bugzilla/blob/master/lib/active_bugzilla/service.rb#L66

Reply:
One other note on accessing Settings directly, is that there is a service level accessor for this which is initialized at boot time.

https://github.com/ManageIQ/miq_bot/blob/5d1537dfb8288f4ed96b7b0907fe73fe836f4988/lib/service_mixin.rb#L11-L14
https://github.com/ManageIQ/miq_bot/blob/5d1537dfb8288f4ed96b7b0907fe73fe836f4988/config/initializers/services_initializer.rb#L2

The original intent was that a service could override the setter, and raise an exception is something was missing, but it seems BugzillaService never had that, instead relying on ActiveBugzilla to do the check.

Reply:
> One other note on accessing Settings directly, is that there is a service-class level accessor for this which is initialized at boot time.

Yeah, I had a bit of trouble getting that to work, but I think that makes a bit more sense on how that works now.  Will look into that.